crates/typst-library/src/foundations/str.rs

```diff
@@ -793,6 +817,18 @@ cast! {
     v: Str => Self::Str(v),
 }
 
+#[derive(PartialEq, Cast)]
```

Comment:
```suggestion
#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash, Cast)]
```
please also add brief documentation to the enum.